File: app/src/main/java/mishka/mingeo/data/model/Borehole.java
package mishka.mingeo.data.model;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.PrimaryKey;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

import static android.arch.persistence.room.ForeignKey.CASCADE;

@Entity
//        (foreignKeys = @ForeignKey(entity = Pumping.class,
//                                    parentColumns = "id",
//                                    childColumns = "pumpingId",
//                                    onDelete = CASCADE))
public class Borehole implements Serializable, DatabaseEntity  {
    @PrimaryKey(autoGenerate = true)
    public int id;

    private int pumpingId;

    private int distanceFromCentral;


    public Borehole() {
    }

    public Borehole(Pumping pumping) {
        this.pumpingId = pumping.getId();
        this.distanceFromCentral = new Random().nextInt();
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getPumpingId() {
        return pumpingId;
    }

    public void setPumpingId(int pumpingId) {
        this.pumpingId = pumpingId;
    }

    public int getDistanceFromCentral() {
        return distanceFromCentral;
    }

    public void setDistanceFromCentral(int distanceFromCentral) {
        this.distanceFromCentral = distanceFromCentral;
    }
}

File: app/src/main/java/mishka/mingeo/data/model/Borehole.kt
package mishka.mingeo.data.model

import android.arch.persistence.room.Entity
import android.arch.persistence.room.ForeignKey
import android.arch.persistence.room.PrimaryKey

import org.joda.time.Instant

import java.io.Serializable
import java.util.ArrayList
import java.util.Random

import android.arch.persistence.room.ForeignKey.CASCADE

@Entity
class Borehole : Serializable{
    @PrimaryKey(autoGenerate = true)
    var id = 0L

    var pumpingId = 0L

    var initialDepth = 0f

    var distanceFromCentral = 0f

    var headHeight = 0f

    constructor() {}

    constructor(pumping: Pumping) {
        this.pumpingId = pumping.id
    }
}
